### DISCUSSION TITLE
0604 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#72] [General]
# 오늘의 TMI + 이모지

|      아라       |   민용   |         정현          |  재훈   | 진현  |       유라       |
| :-------------: | :------: | :-------------------: | :-----: | :---: | :--------------: |
| :shushing_face: | :carrot: | :full_moon_with_face: | :moyai: | :bed: | :baguette_bread: |

# 회의 및 토론

* 뤼이드

  * 데이터를 줄일 수 있는 확인점 . int64=>int32=>int16
  * tag대체 900대로 줄이고 나머지는 컷
  * timestamp는 날리거나 scaling
  * LSTM 베이스라인으로 .73~.74 기준으로 잡고 그걸 넘기는 지 확인하는 걸로 !

* DKVMN

  * 리뷰 : concept별로 Key와 Value로 구분해서 나온 모델. 
  * key를 concept이 담긴 텐서로 실험해봤는데 결과가 그렇게 좋지 않았다.

* SAINT 실험중

  * Feature 추가할 수록 성능이 좋아짐
  * 문제를 푸는데 걸린시간인데 마지막 문제는 어떻게 하지?
  * 평균/최대값 넣어보고 안되면 임의로 넣어보기 

* Last Query 실험중

  * 성능을 올리는데 집중을 할 예정
  * SAINT baseline: 0.7607
    Last Query baseline: 0.7638
    Last Query baseline LB: 0.7579 (차이: 0.0059)
  * cate: nth_test, problem_number
    cont: mean 3가지
    민용이 이야기한거 추가 실험

* 머신러닝 진행사항

  * train, test 합친거로 
  * 피처를 추가해서 
  * 돌렸는데 .83 -> .76으로 떨어짐

* 인수인계[재훈 --> 유라]

  * hold out이 겹치는 부분이 있는줄 알았는데 1개만 겹친다.
  * 겹치는 부분은 없었는데 다른 기준이 필요한 상황이다.

  * pycrate 안에서 fold 나눌 때 validation 기준을 넘겨줄 수 없음
  * 마지막 문제에 대해서만 Valid 셋 나누는 기준을 특정하는게 어려움.
  * 윈도우 형태로 특징을 잡아서 마지막문제 행 기준으로 피처를 생성하자

**코드관련**

* 함수형태로 시간관련 feature를 더 수정
  * 민용) save_pickle, load_pickle 수정해서 올릴 예정
* df넣으면 feature가 생성된 df가 바로 나오도록 수정하기
* 변수명을 snake형태로 수정하기

**[인사이트]**

**모델의 표현력이 어느정도 있을 때 stratified 한 시드변화를 주는것 이 유효할 수 있겠다.**

**현재 대회의 테이터가 적기 때문에 피처를 추가해주는 것이 모델이 학습을 하는데 도움이 되는 것 같다.**

# 저녁메뉴

|   아라   |     민용     |            정현             |      재훈      |    진현     |          유라          |
| :------: | :----------: | :-------------------------: | :------------: | :---------: | :--------------------: |
| 뼈해장국​ | 카레:carrot: | 햄버거:full_moon_with_face: | 카레:question: | 설레드:bed: | 맘모스:baguette_bread: |

**Comment by arabae** [comment]
**주말 계획**
- [ ] [실험] Last Query에 사용된 feature로 변경하여 실험 (최대한 비슷하게)
- [x] [실험] 민용 feature만 사용하여 Last Query 실험
- [x] [실험] Last Query mask 추가 ⭐
- [ ] [실험] conv 등 아이디어 생각
- [ ] [실험] stratified를 위한 key 아이디어 생각

**Comment by gvsteve24** [comment]
**주말계획**

- [x] [실험] Tag처리해서 데이터 개수 1e8/1e7/5e6/1e6으로 구성후 돌리기 
- [ ] [실험] 실험 결과 기록하기
- [x] [강의] transformer tuning, ml pipeline 듣기
- [ ] [과제] Riiid 대회 파트, feature engineering 부분 실습하기
- [ ] [논문] Transformer 읽기

**Comment by MignonDeveloper** [comment]
**주말계획**

- [x] [실험] max_seq_len 1000으로 늘려보기 -> 늘렸는데 거의 비슷하고 시간만 잡아먹는듯
- [x] [실험] Data Augmentation 진행
- [x] [실험] Stratified K-Fold 마지막 문제를 맞췄는지 틀렸는지 기준으로 진행 -> 거의 비슷하게 나오네 random kfold랑,, 그러면 일단 stratified로 가져가자.
- [ ] [실험] 지금까지 실험해온 과정 한번 쭉 정리하기 (노션에서 위키에 복붙)
- [ ] [강의] Transformer 개념 복습
- [ ] [논문] SAINT 논문 읽기

**Comment by Yuuraa** [comment]
주말 계획
- [x] Feature 변수명 바꾸고 함수 안에 넣어서 discussion 올리기
- [ ] 좋은 hold out, validation set을 찾아보자
- [x] Pycaret에 custom KFold 넣어서 돌리기
- [ ] 밀린 강의 좀 보기.... ㅠ

**Comment by KimJinHye0n** [comment]
주말계획
- [ ] 관련 Feature 찾아보기
- [ ] Saint 모델 돌려보기
- [ ] 밀린 강의 보기
